server/gc/src/main/java/org/apache/accumulo/gc/metrics/GcMetrics.java

```diff
@@ -43,88 +50,165 @@
 
   private final SimpleGarbageCollector gc;
 
-  // metrics gauges / counters.
-  private final MutableGaugeLong gcStarted;
-  private final MutableGaugeLong gcFinished;
-  private final MutableGaugeLong gcCandidates;
-  private final MutableGaugeLong gcInUse;
-  private final MutableGaugeLong gcDeleted;
-  private final MutableGaugeLong gcErrors;
-
-  private final MutableGaugeLong walStarted;
-  private final MutableGaugeLong walFinished;
-  private final MutableGaugeLong walCandidates;
-  private final MutableGaugeLong walInUse;
-  private final MutableGaugeLong walDeleted;
-  private final MutableGaugeLong walErrors;
-
-  private final MutableGaugeLong postOpDuration;
-  private final MutableGaugeLong runCycleCount;
-
-  GcMetrics(final SimpleGarbageCollector gc) {
+  public GcMetrics(SimpleGarbageCollector gc) {
     super(jmxName + ",sub=" + gc.getClass().getSimpleName(), description, "accgc", record);
     this.gc = gc;
 
-    MetricsRegistry registry = super.getRegistry();
-
-    gcStarted = registry.newGauge(GC_METRIC_PREFIX + "Started",
-        "Timestamp GC file collection cycle started", 0L);
-    gcFinished = registry.newGauge(GC_METRIC_PREFIX + "Finished",
-        "Timestamp GC file collect cycle finished", 0L);
-    gcCandidates = registry.newGauge(GC_METRIC_PREFIX + "Candidates",
-        "Number of files that are candidates for deletion", 0L);
-    gcInUse =
-        registry.newGauge(GC_METRIC_PREFIX + "InUse", "Number of candidate files still in use", 0L);
-    gcDeleted =
-        registry.newGauge(GC_METRIC_PREFIX + "Deleted", "Number of candidate files deleted", 0L);
-    gcErrors =
-        registry.newGauge(GC_METRIC_PREFIX + "Errors", "Number of candidate deletion errors", 0L);
-
-    walStarted = registry.newGauge(GC_METRIC_PREFIX + "WalStarted",
-        "Timestamp GC WAL collection started", 0L);
-    walFinished = registry.newGauge(GC_METRIC_PREFIX + "WalFinished",
-        "Timestamp GC WAL collection finished", 0L);
-    walCandidates = registry.newGauge(GC_METRIC_PREFIX + "WalCandidates",
-        "Number of files that are candidates for deletion", 0L);
-    walInUse = registry.newGauge(GC_METRIC_PREFIX + "WalInUse",
-        "Number of wal file candidates that are still in use", 0L);
-    walDeleted = registry.newGauge(GC_METRIC_PREFIX + "WalDeleted",
-        "Number of candidate wal files deleted", 0L);
-    walErrors = registry.newGauge(GC_METRIC_PREFIX + "WalErrors",
-        "Number candidate wal file deletion errors", 0L);
-
-    postOpDuration = registry.newGauge(GC_METRIC_PREFIX + "PostOpDuration",
-        "GC metadata table post operation duration in milliseconds", 0L);
-
-    runCycleCount = registry.newGauge(GC_METRIC_PREFIX + "RunCycleCount",
-        "gauge incremented each gc cycle run, rest on process start", 0L);
-
+    metricValues = new AtomicReference<>(this.gc.getGcCycleMetrics());
+
```

Comment:
I don't think this needs to be an atomic reference - there should be only one thread in the constructor.